mm2src/mm2_main/tests/mm2_tests/eth_tests.rs

```diff
@@ -77,3 +114,27 @@ fn test_disable_eth_coin_with_token() {
     // Then try to disable ETH platform coin.
     block_on(disable_coin(&mm, "ETH"));
 }
+
+#[test]
+#[cfg(not(target_arch = "wasm32"))]
+fn test_disable_eth_coin_with_token_without_balance() {
+    let passphrase = get_passphrase(&".env.client", "BOB_PASSPHRASE").unwrap();
+    let coins = json! ([
+        {"coin":"ETH","name":"ethereum","protocol":{"type":"ETH"},"rpcport":80,"mm2":1},
+        {"coin":"JST","name":"jst","rpcport":80,"mm2":1,"protocol":{"type":"ERC20","protocol_data":{"platform":"ETH","contract_address":"0x2b294F029Fde858b2c62184e8390591755521d8E"}}}
+    ]);
+    let conf = Mm2TestConf::seednode(&passphrase, &coins);
+    let mm = block_on(MarketMakerIt::start_async(conf.conf, conf.rpc_password, None)).unwrap();
+    let enable_eth_with_tokens = block_on(enable_eth_with_tokens_without_balance(
+        &mm,
+        "ETH",
+        &["JST"],
+        ETH_DEV_NODES,
+    ));
+
+    let enable_eth_with_tokens: RpcV2Response<EnableEthWithTokensResponse> =
```

Comment:
Avoiding these duplication is a good idea, I am not entirely sure it should be in `mm2_rpc` module though since this is a crate that defines the underlying RPC protocol for all requests and responses and we shouldn't include specific requests/responses there. I would like to get @ozkanonur opinion on this since he has better knowledge of the project/crates structure.
In the end, I think unifying these structs is out of this PR scope and should be done in a separate refactoring PR.

Reply:
I agree about unifying duplicated data-types. But as it's been said, that's out of this PR's scope.

cc https://github.com/KomodoPlatform/atomicDEX-API/issues/1269

Reply:
@shamardy , @ozkanonur. I'm sorry for arguing argue, but let me defend RPC API.
EnableEthWithTokensResponse and its twin EthWithTokensActivationResult were entered with this query, and they are the subject of this query. That is why I encourage you to revise an approach a little. Let's introduce this certain, definite part of `RPC protocol` where and how it should be introduced.

[#1269](https://github.com/KomodoPlatform/atomicDEX-API/issues/1269) exists already a year and presumably will exist a one more year - it does not allow us to breed a zoo of replicas like that. The matter of this issue things that were introduced before but not the things.

Thank you )

Reply:
> I'm sorry for arguing argue, but let me defend RPC API.

We are not against the idea but we need to think about this separately from this PR and in the context of the whole crates structure. Requests and Responses structs are currently scattered across multiple crates and unifying them will require first to choose the best place for them (in a new common crate for requests/responses that is visible across all the project or in an already existing crate) and then the actual refactor might touch on other parts of the code that we are not currently aware of. This PR is for bug fixing week (was last week) and should be closed this week (so we can focus on other tasks) if there is no problems related to what the PR actually supposed to do, that's what I mean by scope.

> https://github.com/KomodoPlatform/atomicDEX-API/issues/1269 exists already a year and presumably will exist a one more year - it does not allow us to breed a zoo of replicas like that. The matter of this issue things that were introduced before but not the things.

It's about priority too, if you think this has higher priority than other tasks, you can pick it up in next bug fixing week or any other time so it will not stay for another year :)

Reply:
mm2_rpc/mm_protocol.rs is an ideal place )

Reply:
If these are used only in places where these are defined that would be ok, these are getting a part of [pulic interface](https://developers.komodoplatform.com/basic-docs/atomicdex-api-20/enable_eth_with_tokens.html#response) though. In fact these are supposed and could be used by a real client and should be exposed in way that provides an easier way to get them. It would be great to keep protocol data types in a `mm_protocol`, and this is a practice I ever met before. I don't mean mm_protocol.rs as it is to store every protocol related struct - I mean keeping protocol datatype structured under mm2_rpc. I will definitely move all the other structures there during working out adex-cli

Concerning `EnableEthWithTokensResponse` if it's utilized in tests, tests should be able to get to the origin one, not to the replica. Replica types are the source of problem and puzzles people who are reading the code. It's called "write everything twice" I'm sure you're aware of it. 
Let me note I'm not speaking about moving the old ones, I'm speaking about not breeding the new ones.

Reply:
> mm2_rpc/mm_protocol.rs is an ideal place ), it won't take more then half an hour

I think we shouldn't rush on the refactoring like that. The current crate structure was applied around ~8-9 months ago. Before, it was much worse. I would like to spend some time and think for proper refactoring rather than moving code between crates continuously. I know the current one is not proper, but applying this kind of changes will not really help on every PR. I personally need more time to think on re-designing this.